java/connector-node/python-client/integration_tests.py

```diff
@@ -41,88 +43,122 @@ def make_mock_schema_stream_chunk():
+def load_json_payload(input_file):
+    sink_input = load_input(input_file)
     payloads = []
-    if use_json:
-        sink_input = load_input(input_file)
-        for batch in sink_input:
-            row_ops = []
-            for row in batch:
-                row_ops.append(connector_service_pb2.SinkStreamRequest.WriteBatch.JsonPayload.RowOp(
-                    op_type=row['op_type'], line=str(row['line'])))
-            payloads.append(connector_service_pb2.SinkStreamRequest.WriteBatch.JsonPayload(
-                row_ops=row_ops))
-    else:
-        sink_input = load_binary_input(input_file)
-        payloads.append(connector_service_pb2.SinkStreamRequest.WriteBatch.StreamChunkPayload(
-            binary_data=sink_input))
+    for batch in sink_input:
+        row_ops = []
+        for row in batch:
+            row_ops.append(
+                connector_service_pb2.SinkStreamRequest.WriteBatch.JsonPayload.RowOp(
+                    op_type=row["op_type"], line=str(row["line"])
+                )
+            )
+
+        payloads.append(
+            {
+                "json_payload": connector_service_pb2.SinkStreamRequest.WriteBatch.JsonPayload(
+                    row_ops=row_ops
+                )
+            }
+        )
+    return payloads
+
+
+def load_stream_chunk_payload(input_file):
+    payloads = []
+    sink_input = load_binary_input(input_file)
+    payloads.append(
+        {
+            "stream_chunk_payload": connector_service_pb2.SinkStreamRequest.WriteBatch.StreamChunkPayload(
+                binary_data=sink_input
+            )
+        }
+    )
     return payloads
 
 
-def test_sink(type, prop, payload_input, use_json, table_schema=make_mock_schema()):
+def test_sink(prop, format, payload_input, table_schema):
     # read input, Add StartSink request
-    request_list = [connector_service_pb2.SinkStreamRequest(start=connector_service_pb2.SinkStreamRequest.StartSink(
-        format=connector_service_pb2.SinkPayloadFormat.JSON if use_json else connector_service_pb2.SinkPayloadFormat.STREAM_CHUNK,
-        sink_config=connector_service_pb2.SinkConfig(
-            connector_type=type,
-            properties=prop,
-            table_schema=table_schema
+    request_list = [
+        connector_service_pb2.SinkStreamRequest(
+            start=connector_service_pb2.SinkStreamRequest.StartSink(
+                format=format,
+                sink_config=connector_service_pb2.SinkConfig(
+                    connector_type=prop["connector"],
+                    properties=prop,
+                    table_schema=table_schema,
+                ),
+            )
         )
-    ))]
+    ]
 
-    with grpc.insecure_channel('localhost:50051') as channel:
+    with grpc.insecure_channel("localhost:50051") as channel:
         stub = connector_service_pb2_grpc.ConnectorServiceStub(channel)
-        epoch = 0
+        epoch = 1
         batch_id = 1
         # construct request
```

Comment:
Should the epoch start from 0 and the batch_id start from 1?

Reply:
There is no such requirement. They just have to be monotonically increasing. When epoch=0, the request pb printed in python will not have epoch printed because 0 is a default value.